internal/datanode/channel_meta.go

```diff
@@ -61,7 +61,7 @@ type Channel interface {
 	getCollectionAndPartitionID(segID UniqueID) (collID, partitionID UniqueID, err error)
 	getChannelName(segID UniqueID) string
 
-	addSegment(req addSegmentReq) error
+	addSegment(ctx context.Context, req addSegmentReq) error
```

Comment:
add ctx for the other functions 🤔

Reply:
It's the only method we need to download files from S3, this's why I added a ctx.

linked a issue here: https://github.com/milvus-io/milvus/issues/27206